File: 4_Pregunta/CuartaPregunta.scala
def calculadora(x:Int, y:Int,funcion:(Int, Int) => Int):Int = funcion(x,y)

def suma(x:Int, y:Int):Int = x+y
def resta(x:Int, y:Int):Int = x-y
def multiplicacion(x:Int, y:Int):Int = x*y
def division(x:Int, y:Int):Int = x/y

val c = calculadora(2,2,multiplicacion)

File: 4_Pregunta/CuartaPregunta.hs
calculadora :: t1 -> t2 (t1 -> t2 -> t3) -> t3
calculadora num1 num2 op = op num1 num2

suma :: Num a => a -> a -> a
suma num1 num2 = num1 + num2

resta :: Num a => a -> a -> a
resta num1 num2 = num1 - num2

multiplicacion :: Num a => a -> a -> a
multiplicacion num1 num2 = num1 * num2

division :: Num a => a -> a -> a
division num1 num2 = num1 / num2
